blik-base/src/main/java/com/adyen/checkout/blik/BlikOutputData.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by josephj on 4/12/2020.
+ */
+
+package com.adyen.checkout.blik;
+
+import android.text.TextUtils;
+
+import androidx.annotation.NonNull;
+
+import com.adyen.checkout.base.component.OutputData;
+import com.adyen.checkout.base.validation.ValidatedField;
+import com.adyen.checkout.core.log.LogUtil;
+import com.adyen.checkout.core.log.Logger;
+
+class BlikOutputData implements OutputData {
+    private static final String TAG = LogUtil.getTag();
+
+    private static final int BLIK_CODE_LENGTH = 6;
+
+    private final ValidatedField<String> mBlikCodeField;
+
+    BlikOutputData(@NonNull String blikCode) {
+        mBlikCodeField = new ValidatedField<>(blikCode, getBlikCodeValidation(blikCode));
+    }
+
+    @Override
+    public boolean isValid() {
+        return mBlikCodeField.isValid();
+    }
+
+    @NonNull
+    public ValidatedField<String> getBlikCodeField() {
+        return mBlikCodeField;
+    }
+
+    private ValidatedField.Validation getBlikCodeValidation(@NonNull String blikCode) {
+        if (TextUtils.isEmpty(blikCode)) {
+            return ValidatedField.Validation.PARTIAL;
```

Comment:
You were asking me why empty field was not showing validation error on the UI, this is probably why. 
Just a UX choice, you can choose to show the error message for partial validation on the View.

Reply:
My question was mainly about after you press the Pay button, but then I found out that the issue only happens in v3 and it seems to be fixed in v4 :+1: